File: CSharp/CSharp Winform/School/CNDOTNET-HaDuc25/BT DOT Net CSharp/Console/test/test/TamGiac.cs
using System;

namespace test
{
    class TamGiac
    {
        #region Khai bao
        //ctrl + r,e
        private double canhA;
        private double canhB;
        private double canhC;

        public double CanhA { get => canhA; set => canhA = value; }
        public double CanhB { get => canhB; set => canhB = value; }
        public double CanhC { get => canhC; set => canhC = value; }
        #endregion

        #region isTriangle 
        public void isTriangle()
        {
            double chuVi = CanhA + CanhB + CanhC;
            double temp = chuVi / 2;
            double dienTich = Math.Sqrt(temp * (temp - CanhA) * (temp - CanhB) * (temp - CanhC));

            Console.WriteLine("Chu vi tam giac: {0}, Dien Tich tam giac: {1}", chuVi, Math.Round(dienTich, 3));

            if (CanhA < CanhB + CanhC && CanhB < CanhA + CanhA + CanhC && CanhC < CanhA + CanhB)
            {
                if (CanhA * CanhA == CanhB * CanhB + CanhC * CanhC || CanhB * CanhB == CanhA * CanhA + CanhC * CanhC || CanhC * CanhC == CanhA * CanhA + CanhB * CanhB)
                {
                    Console.WriteLine("=>Day la tam giac vuong!");
                }
                else if (CanhA == CanhB && CanhB == CanhC)
                {
                    Console.WriteLine("=>Day la tam giac deu!");
                }
                else if (CanhA == CanhB || CanhA == CanhC || CanhB == CanhC)
                {
                    Console.WriteLine("=>Day la tam giac can");
                }
                else
                {
                    Console.WriteLine("=>Day la tam giac thuong");
                }
            }
            else
            {
                Console.WriteLine("Canh A, B, C khong phai la 3 canh cua mot tam giac");
            }
        }
        #endregion

        #region Nhap, Xuat
        public void Nhap()
        {
            Console.Write("Nhap canh A: ");
            this.CanhA = Convert.ToDouble(Console.ReadLine());
            Console.Write("Nhap canh B: ");
            this.CanhB = Convert.ToDouble(Console.ReadLine());
            Console.Write("Nhap canh C: ");
            this.CanhC = Convert.ToDouble(Console.ReadLine());
        }

        public void Xuat()
        {
            Console.WriteLine("Canh A: " + this.CanhA);
            Console.WriteLine("Canh B: " + this.CanhB);
            Console.WriteLine("Canh C: " + this.CanhC);
        }
        #endregion
    }
}

File: Java/OnTap/TamGiac.java
package baitap_th2;

import java.io.*;
import java.util.Scanner;

public class TamGiac {
	private double canh_a, canh_b, canh_c;
	TamGiac(){}
	
	TamGiac(double a, double b, double c)
	{
		canh_a = a;
		canh_b = b;
		canh_c = c;
	}
	
	public void isTriangle()
	{
        double chuVi = canh_a + canh_b + canh_c;
        double temp = chuVi / 2;
        double dienTich = Math.sqrt(temp * (temp - canh_a) * (temp - canh_b) * (temp - canh_c));
        
//        System.out.printf("Chu vi tam giac: %.2f, Dien Tich tam giac: %.2f",chuVi,dienTich);
       
        
        //checking
        if (canh_a < canh_b + canh_c && canh_b < canh_a + canh_a + canh_c && canh_c < canh_a + canh_b)
        {
        	System.out.printf("Dien Tich tam giac: %.2f",dienTich);
            if (canh_a * canh_a == canh_b * canh_b + canh_c * canh_c || canh_b * canh_b == canh_a * canh_a + canh_c * canh_c || canh_c * canh_c == canh_a * canh_a + canh_b * canh_b)
            {
                System.out.println("\t =>Day la tam giac vuong!");
            }
            else if (canh_a == canh_b && canh_b == canh_c)
            {
                System.out.println("\t =>Day la tam giac deu!");
            }
            else if (canh_a == canh_b || canh_a == canh_c || canh_b == canh_c)
            {
                System.out.println("\t =>Day la tam giac can");
            }
            else
            {
                System.out.println("\t =>Day la tam giac thuong");
            }
        }
        else
        {
        	System.out.println("\t =>Canh A, B, C khong phai la 3 canh cua mot tam giac");
        }
	}
	
	
	
	//nhap
	public void nhap() 
	{
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Nhap canh A: ");
		do {
			try {
				canh_a = Double.parseDouble(sc.nextLine());
			} catch (Exception e) {
				canh_a = 0;
			}
			if (canh_a == 0)
				System.out.print("\t Nhap sai => Nhap lai canh A: ");
		} while (canh_a <= 0);
		
		System.out.print("Nhap canh B: ");
		do {
			try {
				canh_b = Double.parseDouble(sc.nextLine());
			} catch (Exception e) {
				canh_b = 0;
			}
			if (canh_b == 0)
				System.out.print("\t Nhap sai => Nhap lai canh B: ");
		} while (canh_b <= 0);
		
		System.out.print("Nhap canh C: ");
		do {
			try {
				canh_c = Double.parseDouble(sc.nextLine());
			} catch (Exception e) {
				canh_c = 0;
			}
			if (canh_c == 0)
				System.out.print("\t Nhap sai => Nhap lai canh C: ");
		} while (canh_c <= 0);
	}
	
	//xuat
	public void xuat() 
	{
		System.out.println("\n <====== Result ======> \n");
		
		System.out.println("Canh A: " + canh_a); 
		System.out.println("Canh B: " + canh_b);  
		System.out.println("Canh C: " + canh_c); 
	}
	
	
	public static void main(String[] args) {
		TamGiac tmp = new TamGiac();
		System.out.println("Tam giac :> ");
		tmp.nhap();
		tmp.xuat();
		tmp.isTriangle();
		
	}
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
}
